internal/tpm/tpm_test.go

```diff
@@ -0,0 +1,123 @@
+package tpm
+
+import (
+	"crypto/rand"
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+	"testing"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm/legacy/tpm2"
+	"github.com/stretchr/testify/require"
+)
+
+// These unit tests all use the tpm simulator from go-tpm-tools.
+
+type TestFixture struct {
+	tpm *TPM
+}
+
+func setupTestFixture(t *testing.T) *TestFixture {
+	t.Helper()
+
+	tpm, err := OpenTPMSimulator()
+	if err != nil {
+		t.Errorf("unable to open tpm simulator")
+	}
+
+	return &TestFixture{tpm: tpm}
+}
+
+func setupTestData(t *testing.T) (*TPM, *client.Key, []byte, *tpm2.PCRSelection) {
+	t.Helper()
+
+	f := setupTestFixture(t)
+
+	key, err := f.tpm.CreateLAK()
+	if err != nil {
+		t.Errorf("unable to create local attestation key")
+	}
+
+	nonce := make([]byte, 8)
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		t.Errorf("failed to create nonce: %v", err)
```

Comment:
same here you need to stop progress when there is an error minimally. either return a proper error or return after the log